### ISSUE TITLE
Upgrade from 21.04 to 21.10

### ISSUE THREAD

**Issue by ikohlernz** [pop-os/pop#2059]
**Distribution**:

NAME="Pop!_OS"
VERSION="21.04"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.04"
VERSION_ID="21.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute
LOGO=distributor-logo-pop-os

pop-upgrade: calling FetchUpdates method failed
  caused by: failed to exec `apt-get full-upgrade --print-uris`: status is unknown: exit status: 100

apt-get full-upgrade --print-uris
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 network-manager : Depends: libnm0 (= 1.30.0-1ubuntu3) but 1.32.12-0ubuntu1 is installed
                   Recommends: ppp but it is not installed
                   Recommends: modemmanager but it is not installed
 network-manager-l2tp : Depends: ppp (< 2.4.7-3~) but it is not installed
                        Depends: ppp (>= 2.4.7-2+~) but it is not installed
 network-manager-pptp : Depends: ppp (< 2.4.7-3~) but it is not installed
                        Depends: ppp (>= 2.4.7-2+~) but it is not installed
 pptp-linux : Depends: ppp but it is not installed
 xl2tpd : Depends: ppp but it is not installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).

apt policy network-manager
network-manager:
  Installed: 1.30.0-1ubuntu3
  Candidate: 1.30.0-1ubuntu3
  Version table:
 *** 1.30.0-1ubuntu3 500
        500 http://us.archive.ubuntu.com/ubuntu hirsute/main amd64 Packages
        100 /var/lib/dpkg/status
    
 pop-os:~$ apt policy ppp
ppp:
  Installed: (none)
  Candidate: 2.4.7-2+4.1ubuntu8
  Version table:
     2.4.9-1+1ubuntu1 -1
        100 /var/lib/dpkg/status
     2.4.7-2+4.1ubuntu8 500
        500 http://us.archive.ubuntu.com/ubuntu hirsute/main amd64 Packages

       
pop-os:~$ apt policy libnm0
libnm0:
  Installed: 1.32.12-0ubuntu1
  Candidate: 1.32.12-0ubuntu1
  Version table:
 *** 1.32.12-0ubuntu1 100
        100 /var/lib/dpkg/status
     1.30.0-1ubuntu3 500
        500 http://us.archive.ubuntu.com/ubuntu hirsute/main amd64 Packages

pop-os:~$ apt policy network-manager-pptp
network-manager-pptp:
  Installed: 1.2.8-3
  Candidate: 1.2.8-3
  Version table:
 *** 1.2.8-3 500
        500 http://us.archive.ubuntu.com/ubuntu hirsute/main amd64 Packages
        100 /var/lib/dpkg/status

**Issue/Bug Description:** After trying to update to 21.10 via the gui pop up,  it never finished. Might have been interrupted during the process.  Laptop rebooted and afterwards trying to update anything fails, I can now see dependency errors, tried a number of things, but cannot get pass it. 
Tried suggested fix:

sudo apt clean
sudo apt update -m
sudo dpkg --configure -a
sudo apt install -f
sudo apt dist-upgrade
sudo apt autoremove --purge

But any apt get command fails now.

Similar issue also reported during beta testing by someone else : https://githubmemory.com/repo/pop-os/beta/issues/238

**Steps to reproduce:**

Have Pop 21.04 up to date
Tried to download/upgrade the PopOS21.10 that never finished.
Then facing the dependency error.

Expected behavior:

Pop! OS is upgraded from 21.04 to 21.10.

**Comment by SuluCiprian** [comment]
I am having the same issue.

**Comment by leviport** [comment]
Due to the overwhelming demand for 21.10, we have had to expand our servers' bandwidth. If you are experiencing any connection errors please follow the instruction here: https://support.system76.com/articles/upgrade-pop/#repairing-upgrade-errors

**Comment by ikohlernz** [comment]
Same error after following the repair upgrade  link:

sudo rm /etc/apt/sources.list.d/pop-os-ppa.sources
sudo systemctl restart pop-upgrade
sudo pop-upgrade release upgrade

checking if pop-upgrade requires an update
Current Release: 21.04
Upgrading to: 21.10
New version available: 3
Event: updating package lists
Event: fetching updated packages for the current release
Release upgrade status: release upgrade aborted: failed to fetch apt URIs to fetch
pop-upgrade: calling ReleaseUpgradeFinalize method failed
  caused by: release upgrade cannot be finalized, because a release upgrade was not performed

**Comment by pjkaufman** [comment]
I am hitting this issue as well. I cannot tell from what has already been commented, but it looks like the issue breaks the ability to install packages or updates. When I tried to do updates via `sudo apt upgrage` I got an error saying dependencies were broke. So I ran the suggested command of `sudo apt --fix-broken install` and got the following error:

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Correcting dependencies... failed.
The following packages have unmet dependencies:
 network-manager : Depends: libnm0 (= 1.30.0-1ubuntu3) but 1.32.12-0ubuntu1 is installed
                   Breaks: ppp (>= 2.4.7-3~) but 2.4.9-1+1ubuntu1 is installed
 network-manager-pptp : Depends: ppp (< 2.4.7-3~) but 2.4.9-1+1ubuntu1 is installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
E: Unable to correct dependencies

I am not sure how to proceed since everything I am seeing leads me to believe that there was a bad installation and it cannot rollback the change, but it also is not  able to fix it either.

Any help on figuring out how to get past this issue and being able to upgrade to the latest version would be superb!

**Comment by srinjoychakravarty** [comment]
Have the exact same issues as the others on the a system76 mmachine after the failed firmware upgrade:

```sh
$ sudo su
root@pop-os:/home/eos# apt --fix-broken install
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Correcting dependencies... failed.
The following packages have unmet dependencies:
 network-manager : Depends: libnm0 (= 1.30.0-1ubuntu3) but 1.32.12-0ubuntu1 is installed
                   Breaks: ppp (>= 2.4.7-3~) but 2.4.9-1+1ubuntu1 is installed
 network-manager-pptp : Depends: ppp (< 2.4.7-3~) but 2.4.9-1+1ubuntu1 is installed
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
E: Unable to correct dependencies
```

**Comment by nanoharsha** [comment]
Having the same issue at my end too !! 
I was trying to upgrade, and the system got hung at System Upgrade complete, Rebooting now. When I didn't see any progress, I force rebooted and stuck with this error message. I am using a Dell XPS machine. 
Tried all steps mentioned in this [article](https://support.system76.com/articles/pop-incomplete-upgrade), but unable to resolve. Pls find attached screenshots.

![Synaptic](https://user-images.githubusercontent.com/10630319/146685651-214b50c0-57dc-4eae-a136-2ffdb3d97fad.png)
![About](https://user-images.githubusercontent.com/10630319/146685654-d49523ca-da55-4765-a8c7-f3db9dfde952.png)
![Error_message_Upgrading](https://user-images.githubusercontent.com/10630319/146685655-74f80e2d-aaf7-41f4-8bef-641121a835a8.png)

**Comment by crubel** [comment]
Just leaving a quick note here, I attempted to upgrade my Oryx7 last night.  Everything appeared to go ok with upgrade and initial reboot, logged in and everything was ok until next reboot, now I have a dark grey screen no login, no cursor..  
Any suggestions?  Need more info let me know how to get it for you..

I will attempt a fresh install after ensuring my backup is good..I find it hard to believe I can't get this working properly on a System76 Oryp7 from the PopOS company itself, these guys have been great so far, so I'll carry on until it's up again and if not I'll use my backup and restore the 21.04 image I made until these bugs are ironed out...

**Comment by nathanRamaNoodles** [comment]
TL:DR; I got all my stuff successfully transferred to the new installation, and everything works great 👍.

Personally, I've never had Pop-OS updates work properly without some bug rising up.  What works flawlessly is rsync-ing the home folder to the new distro.  Here are the steps I did:

1. Open terminal in old OS
2. Use `rsync` and backup current home folder to an external SSD (SSD takes 10-20 minutes, Hard drive takes hours)
   + `rsync -av /home/ /media/$USER/<external_disk_drive_name>`
3. Then, wipe the old OS and install the new distro/Pop-OS
4. Login into new OS and open terminal
5. Use `rsync` again to restore old `home` folder with proper permissions
   + `rsync -av /media/$USER/<external_disk_drive_name> /home/`

**Comment by pretendasaur** [comment]
Having this same issue which also comes up when trying to install any package. I cannot seem to install the missing dependencies for ppp because the error for libnm0 comes up. Trying to install the package for libnm0 comes up with the error for ppp not being installed.

**Comment by jacobgkau** [comment]
https://github.com/pop-os/upgrade/issues/238 should prevent this issue for anyone who did not attempt the upgrade prior to the 18th.

The easiest way to recover from this dependency issue would be doing a refresh install from either the Recovery partition or a 21.10 [live disk](https://pop.system76.com/). A refresh install will reset packages (system-wide apps) to the 21.10 defaults, but will preserve all files in your home directory.

For anyone who would rather solve the dependency issue, you need to read the error messages and create an apt command to resolve the dependency problems. For example, some of you have posted a message containing this line:

```
 network-manager : Depends: libnm0 (= 1.30.0-1ubuntu3) but 1.32.12-0ubuntu1 is installed
                   Breaks: ppp (>= 2.4.7-3~) but 2.4.9-1+1ubuntu1 is installed
```

So your apt command would need to include:

```
sudo apt install ppp=2.4.7-2+4.1ubuntu8 libnm0=1.30.0-1ubuntu3
```

The error message should show if any other packages need different versions; add them to the apt command until the dependencies are all resolved.

@crubel Like I said in the other issue where you commented, please open a support ticket at https://support.system76.com/ if you own System76 hardware. You can get one-on-one support there.

**Comment by pebongue** [comment]
Hello, I am having the same issue and I have tried everything suggested above, and still it is not fixed. I do not own a System76 hardware, hence I am not able to lodge a ticket on their website. Please if anyone has managed to solve this issue, let us know.

**Comment by jacobgkau** [comment]
> I have tried everything suggested above, and still it is not fixed.

Where did it stop working? What was the last error message from apt that you got after following those instructions to create an `apt install` command?

The Pop!_Chat is the best place to ask for community support; I helped another user solve it there: https://chat.pop-os.org/pop-os/pl/gcntgbbdopd47ehfx5wfmqmh9c

**Comment by pebongue** [comment]
> > I have tried everything suggested above, and still it is not fixed.
> 
> Where did it stop working? What was the last error message from apt that you got after following those instructions to create an `apt install` command?
> 
> The Pop!_Chat is the best place to ask for community support; I helped another user solve it there: https://chat.pop-os.org/pop-os/pl/gcntgbbdopd47ehfx5wfmqmh9c

After running the command as suggested, everything went to normal. The system upgraded and all updates are working now. Thank you again @jacobgkau

**Comment by markuskreitzer** [comment]
No solution for me either. Sad to say, after being a loyal PopOS user, I'm headed back to Ubuntu land.

**Comment by hillbyte** [comment]
```bash
Release upgrade status: release upgrade aborted: fetch of package failed: package fetching failed

Caused by:
    0: failed to request package
    1: the server made an unrecoverable HTTP protocol violation
    2: [52] Server returned nothing (no headers, no data)
pop-upgrade: calling ReleaseUpgradeFinalize method failed
  caused by: release upgrade cannot be finalized, because a release upgrade was not performed
````
I tried to repair upgrade errors by
`sudo rm /etc/apt/sources.list.d/pop-os-ppa.sources`

tried several times but it didn't work

What should I do to fix this?

**Comment by pebongue** [comment]
> No solution for me either. Sad to say, after being a loyal PopOS user, I'm headed back to Ubuntu land.

Hi @markuskreitzer what did you try so long? What did you get as output? Perhaps we can try to assist from there. Keep well!

**Comment by chandan0000** [comment]
@jacobgkau thank you!

**Comment by AEFOUNDRY** [comment]
> [pop-os/upgrade#238](https://github.com/pop-os/upgrade/issues/238) should prevent this issue for anyone who did not attempt the upgrade prior to the 18th.
> 
> The easiest way to recover from this dependency issue would be doing a refresh install from either the Recovery partition or a 21.10 [live disk](https://pop.system76.com/). A refresh install will reset packages (system-wide apps) to the 21.10 defaults, but will preserve all files in your home directory.
> 
> For anyone who would rather solve the dependency issue, you need to read the error messages and create an apt command to resolve the dependency problems. For example, some of you have posted a message containing this line:
> 
> ```
>  network-manager : Depends: libnm0 (= 1.30.0-1ubuntu3) but 1.32.12-0ubuntu1 is installed
>                    Breaks: ppp (>= 2.4.7-3~) but 2.4.9-1+1ubuntu1 is installed
> ```
> 
> So your apt command would need to include:
> 
> ```
> sudo apt install ppp=2.4.7-2+4.1ubuntu8 libnm0=1.30.0-1ubuntu3
> ```
> 
> The error message should show if any other packages need different versions; add them to the apt command until the dependencies are all resolved.
> 
> @crubel Like I said in the other issue where you commented, please open a support ticket at https://support.system76.com/ if you own System76 hardware. You can get one-on-one support there.

I tried this method and what I am receiving now is this error
``
Package libnm0 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  libnm0:i386

Package ppp is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Version '2.4.7-2+4.1ubuntu8' for 'ppp' was not found
E: Version '1.30.0-1ubuntu3' for 'libnm0' was not found
``

**Comment by jacobgkau** [comment]
> I tried this method and what I am receiving now is this error
> 
> ```
> Package libnm0 is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> However the following packages replace it:
>   libnm0:i386
> 
> Package ppp is not available, but is referred to by another package.
> This may mean that the package is missing, has been obsoleted, or
> is only available from another source
> 
> E: Version '2.4.7-2+4.1ubuntu8' for 'ppp' was not found
> E: Version '1.30.0-1ubuntu3' for 'libnm0' was not found
> ```

Your system must not be having the same issue as others in this thread, so the exact same commands won't work. I can see from the [Ubuntu package tracker](https://packages.ubuntu.com) that both `ppp` and `libnm0` are available for Impish/21.10, but they have newer version numbers than the ones used in the command (which was meant for a Hirsute/21.04 system.)

The output of `sudo apt update` would be necessary to see which repositories your system is currently checking for packages. If the main Ubuntu archive is present and the version is `impish`, then the version numbers would change:

```
sudo apt install ppp=2.4.9-1+1ubuntu1 libnm0=1.32.12-0ubuntu1
```

But that may not help at all unless those are the exact packages your system is having an issue with.

Did you try performing a refresh install from live media like I mentioned earlier?

**Comment by tomxnodes** [comment]
I'm at almost the same exact result as @jacobgkau here
running the suggested snippet works, but then goes down a rabbit hole of needing more dependencies mapped. I'm not sure how to determine those. I found this string in a related ticket:
 sudo apt install ppp=2.4.7-2+4.1ubuntu8 libnm0=1.30.0-1ubuntu3

Those versions got me further, but then running 
 pop-upgrade release upgrade
Starts to work and dies here:
"Release upgrade status: release upgrade aborted: failed to fetch apt URIs to fetch"

Really don't want to have to reconfigure after a live disk install, any further suggestions welcome before I go that way.

**Comment by jacobgkau** [comment]
> Really don't want to have to reconfigure after a live disk install, any further suggestions welcome before I go that way.

Performing a _refresh install_ will keep any personal files and preferences you have configured in your user account, as I have mentioned a few times. You will need to reinstall any non-default packages (except for user-installed Flatpaks), but the preferences for those applications (assuming they're stored the standard way) will still be there.

> but then goes down a rabbit hole of needing more dependencies mapped. I'm not sure how to determine those.

You determine them by seeing which package names the `apt install` command "complains about," then running `apt policy` on those package names. If you provide the actual output you are seeing, others can help you identify the necessary information.

> Those versions got me further, but then running
> pop-upgrade release upgrade
> Starts to work and dies here:

I would recommend not running `pop-upgrade` again until `apt` and `dpkg` are no longer presenting any error messages.

> "Release upgrade status: release upgrade aborted: failed to fetch apt URIs to fetch"

For this, the output of `sudo apt update` from your system may show where the issue lies.

**Comment by jobenvil** [comment]
@jacobgkau 

> Performing a refresh install will keep any personal files and preferences you have configured in your user account, as I have mentioned a few times. You will need to reinstall any non-default packages (except for user-installed Flatpaks), but the preferences for those applications (assuming they're stored the standard way) will still be there.

means that all the dockers and volumens will be keep untouched? I keep all stuff under `/usr/local/etc` and almost nothing inside `/home/myuser`. Is there a link where to read more what exactly will be deleted? . This will be my second unsuccesfull upgrade with pop-os and I'm really thinking again to move to debian/ubuntu again.

**Comment by coolarnie** [comment]
yep exactly the same issue when upgrading from 21.04 to 21.10
Have tried the following suggested [fix](https://support.system76.com/articles/upgrade-pop/#repairing-upgrade-errors)
No joy still.
```
checking if pop-upgrade requires an update
Current Release: 21.04
Upgrading to: 21.10
New version available: 3
Recovery event: fetching recovery files
Event: updating package lists
Event: fetching updated packages for the current release
Release upgrade status: release upgrade aborted: failed to fetch apt URIs to fetch
pop-upgrade: calling ReleaseUpgradeFinalize method failed
  caused by: release upgrade cannot be finalized, because a release upgrade was not performed
```

**Comment by jacobgkau** [comment]
> means that all the dockers and volumens will be keep untouched? I keep all stuff under `/usr/local/etc` and almost nothing inside `/home/myuser`.

@jobenvil Anything outside of the home directory will be reset, so if you know that you have data outside of the home directory, you should back it up first. (Backups are recommended in any case, of course.) If backing up/restoring would be a greater inconvenience than troubleshooting the dependency issue, then that would be a situation where it makes sense to troubleshoot the dependency issue.

> This will be my second unsuccesfull upgrade with pop-os and I'm really thinking again to move to debian/ubuntu again.

I am sorry that you've experienced an issue. You have made one comment on this issue page, which did not contain any output from your system to demonstrate what's going wrong or what might be different from the others who have reported solving the issue on their systems. I did see that you joined the [Mattermost server](https://chat.pop-os.org/pop-os) and provided output from your system, so I'll reply to you there.

I also want to mention that if you're comfortable using Debian, the same upgrade process that Debian uses (manually updating the apt sources lists and performing a dist-upgrade/full-upgrade) also works on Ubuntu and Pop!_OS.

> yep exactly the same issue when upgrading from 21.04 to 21.10

@coolarnie Have you read the rest of the comments in this issue and attempted any of the suggested troubleshooting actions? Please provide the output of `sudo apt update`, `sudo apt full-upgrade`, `sudo apt install -f`, and `apt policy ppp libnm0 network-manager`.

---

This issue tracker is meant to hold information relevant to development. If anyone has steps to recreate this issue from a fresh installation of Pop!_OS, or a list of customizations you've performed to your system to help the developers recreate those steps, those would be the most valuable pieces of information for determining the actual cause of the issue and preventing it for future users/upgrades.

**Comment by jobenvil** [comment]
@jacobgkau Thank you very mich for your support behind the scene. Pop!_OS 🚀 🚀 🚀

**Comment by alvinfrancis** [comment]
@jacobgkau Also had the same issues here, and tracking down the necessary versions with `apt policy` indeed did the trick.  

In the interest of reproducibility and context though, my situation follows.  I was previously on 20.10 when it was already EOL. Trying to upgrade to 21.04 failed due to an issue where apt sources were trying to still update using `groovy` when the Release file was no longer available.  I updated sources manually to point to `hirsute` and proceeded with a `sudo apt full-upgrade` and rebooted.  Trying to update afterwards failed due to this issue.

Hope this information helps.

**Comment by zixxus** [comment]
Hello,
After upgrade system from 21.04 to 21.10 I had that error (when I try to run apt upgrade):

The following packages have unmet dependencies:
 gnome-settings-daemon : Breaks: libmutter-7-0 (< 3.39) but 3.38.4-1ubuntu2pop1~1623339725~21.04~0a2b3e1 is to be installed
E: Broken packages

And with full-upgrade:
The following packages have unmet dependencies:
 pop-desktop : Depends: chrome-gnome-shell but it is not going to be installed
               Depends: gdm3 but it is not going to be installed
               Recommends: gnome-calendar but it is not going to be installed
               Recommends: gnome-contacts but it is not going to be installed
               Recommends: io.elementary.sideload but it is not installable
               Recommends: gnome-remote-desktop but it is not going to be installed
               Recommends: gnome-shell-extension-prefs but it is not going to be installed

And with sudo apt-get upgrade:
The following packages will be DOWNGRADED:
  accountsservice amd-ppt-bin appstream-data-pop appstream-data-pop-icons apt
  apt-transport-https apt-utils com.github.donadigo.eddy debconf debconf-i18n
  firmware-manager-notify firmware-manager-shared flatpak fwupd fwupd-signed
  gir1.2-accountsservice-1.0 gir1.2-flatpak-1.0 gir1.2-goa-1.0
  gir1.2-granite-1.0 gnome-control-center-faces gnome-online-accounts
  gnome-shell-extension-desktop-icons-ng gnome-shell-extension-system76-power
  gnome-terminal gnome-terminal-data hidpi-daemon kernelstub
  libaccountsservice0 libapt-pkg6.0 libdrm-amdgpu1 libdrm-amdgpu1:i386
  libdrm-common libdrm-intel1 libdrm-intel1:i386 libdrm-nouveau2
  libdrm-nouveau2:i386 libdrm-radeon1 libdrm-radeon1:i386 libdrm2 libdrm2:i386
  libegl-mesa0 libfirmware-manager libflatpak-dev libflatpak0 libfwupd2
  libfwupdplugin1 libgbm1 libgbm1:i386 libgl1-mesa-dri libgl1-mesa-dri:i386
  libglapi-mesa libglapi-mesa:i386 libglx-mesa0 libglx-mesa0:i386
  libgoa-1.0-0b libgoa-1.0-common libgoa-backend-1.0-1 libgranite-common
  libnvidia-cfg1-470 libnvidia-common-470 libnvidia-compute-470
  libnvidia-compute-470:i386 libnvidia-decode-470 libnvidia-decode-470:i386
  libnvidia-encode-470 libnvidia-encode-470:i386 libnvidia-extra-470
  libnvidia-fbc1-470 libnvidia-fbc1-470:i386 libnvidia-gl-470
  libnvidia-gl-470:i386 libnvidia-ifr1-470 libnvidia-ifr1-470:i386 libosmesa6
  libosmesa6:i386 libpop-desktop-widget libpop-theme-switcher
  libs76-hidpi-widget libudev1 libudev1:i386 libxatracker2 linux-firmware
  linux-generic linux-headers-5.15.8-76051508
  linux-headers-5.15.8-76051508-generic linux-headers-generic
  linux-image-5.15.8-76051508-generic linux-image-generic linux-libc-dev
  linux-modules-5.15.8-76051508-generic linux-system76 mesa-va-drivers
  mesa-vdpau-drivers mesa-vulkan-drivers mesa-vulkan-drivers:i386
  nautilus-extension-gnome-terminal network-manager-config-connectivity-pop
  nvidia-compute-utils-470 nvidia-dkms-470 nvidia-driver-460 nvidia-driver-470
  nvidia-kernel-common-470 nvidia-kernel-source-470 nvidia-utils-470
  plymouth-theme-pop-basic pop-cosmic pop-fonts pop-gnome-initial-setup
  pop-gnome-shell-theme pop-gtk-theme pop-shell-shortcuts pop-shop
  pop-transition pop-wallpapers python3-repolib python3-update-manager repoman
  sessioninstaller system76-dkms system76-io-dkms system76-power touchegg
  ubuntu-drivers-common ubuntu-release-upgrader-core udev update-manager-core
  update-notifier-common xserver-xorg-video-nvidia-470



My sourcelist use hirsute label, do you know whats is wrong? Shouldn't source.list change to something else? If so, to what?
